homeassistant/components/sensor/airvisual.py

```diff
@@ -95,60 +94,63 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
     if city and state and country:
         _LOGGER.debug(
             "Using city, state, and country: %s, %s, %s", city, state, country)
+        location_id = city + state + country
         data = AirVisualData(
             pav.Client(api_key), city=city, state=state, country=country,
             show_on_map=show_on_map)
     else:
         _LOGGER.debug(
             "Using latitude and longitude: %s, %s", latitude, longitude)
+        location_id = latitude + longitude
         data = AirVisualData(
             pav.Client(api_key), latitude=latitude, longitude=longitude,
             radius=radius, show_on_map=show_on_map)
 
+    unique_id = c_size_t(hash(location_id)).value
+
     data.update()
     sensors = []
     for locale in monitored_locales:
         for sensor_class, name, icon in SENSOR_TYPES:
-            sensors.append(globals()[sensor_class](data, name, icon, locale))
+            sensors.append(globals()[sensor_class](
```

Comment:
We should not use `globals`

Reply:
I chatted about this with @MartinHjelmare a while back; if I were to store references to the actual classes, I think I'd have to define all of them above `setup_platform` (which I didn't feel great about, given that no other component/platform does this). Thoughts?

Reply:
New push incoming – I was able to use references to the actual classes (rather than string representations), but had to move `SENSOR_TYPES` to the bottom. Are you okay with that? I can't find an example of another platform that does this...